src/templates/partials/caddy.hbs

```diff
@@ -1,43 +1,31 @@
 # {{output.header}}
 # {{{output.link}}}
-{{#unless (includes "old" form.config)}}
+{{#unless (minver "2.0.0" form.serverVersion)}}
+# note that Caddy version 1 reached end-of-life > 4 years ago back in 2020
+{{else}}
+{{#if (includes "old" form.config)}}
+# note that Caddy supports only TLSv1.2 and later
+{{else}}
 # note that Caddy automatically configures safe TLS settings
+{{/if}}
 {{/unless}}
 
 # replace example.com with your domain name
-example.com
-{{! This is a big of a kludge due to Caddy restrictions on TLS versions and cipher suites }}
-{{#if (includes "old" form.config)}}
-
-tls {
-    protocols tls1.0 tls1.3
-    {{#if (minver "2.0.0" form.serverVersion)}}
-    ciphers {{{join output.ciphers " "}}}
-    {{ else }}
-    ciphers ECDHE-ECDSA-AES128-GCM-SHA256 ECDHE-RSA-AES128-GCM-SHA256 ECDHE-ECDSA-AES256-GCM-SHA384 ECDHE-RSA-AES256-GCM-SHA384 ECDHE-ECDSA-WITH-CHACHA20-POLY1305 ECDHE-RSA-WITH-CHACHA20-POLY1305 ECDHE-ECDSA-AES128-CBC-SHA ECDHE-RSA-AES256-CBC-SHA ECDHE-RSA-AES128-CBC-SHA ECDHE-ECDSA-AES256-CBC-SHA RSA-AES128-CBC-SHA RSA-AES256-CBC-SHA RSA-3DES-EDE-CBC-SHA
-    {{/if}}
-}
-{{/if}}
-{{#if (includes "intermediate" form.config)}}
+example.com {
 
-# Due to a lack of DHE support, you -must- use an ECDSA cert to support IE 11 on Windows 7
-tls {
-    protocols tls1.2 tls1.3
-    {{#if (minver "2.0.0" form.serverVersion)}}
+  tls {
+{{#if output.ciphers.length}}
+    # Due to a lack of DHE support, you -must- use an ECDSA cert to support IE 11 on Windows 7
     ciphers {{{join output.ciphers " "}}}
-    {{ else }}
-    ciphers ECDHE-ECDSA-AES128-GCM-SHA256 ECDHE-RSA-AES128-GCM-SHA256 ECDHE-ECDSA-AES256-GCM-SHA384 ECDHE-RSA-AES256-GCM-SHA384 ECDHE-ECDSA-WITH-CHACHA20-POLY1305 ECDHE-RSA-WITH-CHACHA20-POLY1305
-    {{/if}}
-}
 {{/if}}
 {{#if (includes "modern" form.config)}}
-
-tls {
     protocols tls1.3
```

Comment:
TLS 1.2 is still considered secure, especially with Caddy's default cipher list. Is it really worth suggesting to break compat with many clients with this option? Caddy's default are already "modern".

Reply:
The naming does not come from "general hardening" or similar, the tool only reflects what is published in https://wiki.mozilla.org/Security/Server_Side_TLS

Reply:
Fair enough :+1:

Reply:
TLSv1.3 is widely supported by modern clients.  Unless sites must support older clients, new configurations can choose TLSv1.3-only, and as @janbrasna pointed out, these are the recommendations for the https://wiki.mozilla.org/Security/Server_Side_TLS "Modern" config.

In January, I plan to release lighttpd with the lighttpd default to use MinVersion TLSv1.3 unless otherwise configured.  This behavior change has been announced on lighttpd releases throughout 2024. 

However, I'll also note that cloudflare "Modern" recommendation still recommends both TLSv1.2 and TLSv1.3
https://developers.cloudflare.com/ssl/edge-certificates/additional-options/cipher-suites/recommendations/

Reply:
We might consider including links to the specific "Modern", "Intermediate", "Old" specifications (including version) in the generated output.

Reply:
I share your concern between HTTP servers and MTAs and other clients.  Instead of differentiating by types of services, we might have a "Modern" and a "Modern-Strict" guideline, where "Modern" is TLSv1.2 and TLSv1.3 without DHE suites, and "Modern-Strict" is TLSv1.3 only.  I think we should discuss this further and try to push our (outsider) recommendations into the security team at Mozilla.

Reply:
Basically MTAs need different profiles for submission than for delivery etc., and even the current intermediate is not enough to support the reality in the wild… so we'll have to figure out a way how to support non-http server universe, while not restricting the development of the specs. One side would be the recommendations proper, that can add the rationale for different interop scenarios, but at the same time the config tool will need to surface that info too. TBD.